### PR TITLE
feat(trainer): add internal_medicine_debug_mode to skip per-unit metric families

### PR DESCRIPTION
**Pull request by yuweivvv**
### PR Category
Trainer / internal medicine monitors

### PR Types
Improvements

### Description

A production run no longer collects the internal-medicine metric families that emit **one curve per structural unit**:

- `moe_health` per-expert shares — `expert_{token,weight}_share_e{i}`
- `mhc_health` per-cell mixing-matrix entries — `{attn,mlp}_h_res_cell{i}`
- `mhc_health` per-stream gate values — `{attn,mlp}_h_{pre,post}_idx{i}`

On a 43-layer mHC + MoE config those are **1392 of 3732 keys per step (37.3%)**, so they dominate the cross-rank aggregation payload, the monitor buffers and the jsonl on disk, while the rest of the schema only grows with depth. Excluded families never enter the metric schema, so this cuts communication and memory, not just the printed keys.

The per-layer aggregates of the same quantities stay on (`h_res_logits_max`, `h_post_mean`, `h_post_stream_concentration`, ...), and those are what the diagnosis layer reads — a cross-check against `diagnose.py` finds none of the excluded metric names referenced there. So the new default costs payload only.

```yaml
internal_medicine_monitors: "qk_stats,moe_health,massive_act,mhc_health,vha_health"
internal_medicine_monitor_interval: 200
internal_medicine_debug_mode: false   # true = collect the per-unit families too
```

The trainer-side surface is deliberately **one bit** rather than a family spec string: which families debug mode adds is a property of the taxonomy, so the set lives beside it as `internal_medicine.core.metric_families.DEBUG_ONLY_FAMILIES` and is pinned by that project's tests. Nothing to typo in a yaml, and changing what "debug" means is a reviewable diff instead of a config sweep.

Changes: `internal_medicine_debug_mode: bool = False` in `PreTrainingArguments`, passed through `Trainer` to `InternalMedicineCallback`, which turns it into the exclusion mapping via `exclusions_for(debug_mode)` at `on_train_begin`.

Requires the family support on the internal_medicine side (bo-ke/llm-internal-medicine#64 plus the `DEBUG_ONLY_FAMILIES` follow-up #65). The import lives inside the existing guarded `try`, so an older library version falls into the "not importable" warning path rather than crashing.

### CODE REVIEW

**Comment by risemeup1111** [comment]
## Failed CI看板

| 流水线名称 | 问题标签 | 修复建议 | 日志片段 |
|------------|----------|----------|----------|
| [33258690340](https://github.com/PaddlePaddle/PaddleFormers/actions/runs/33258690340) | 未解析 | 当前 PR head 未发现失败流水线，本轮未进行 CI 日志分析。 | 未提供 |

<details>
<summary>日志分析报告</summary>

当前 PR head 未发现失败流水线，本轮未进行 CI 日志分析。

</details>

Powered by Nyanpasu with gpt-5.6-luna 默认推理级别, please check the suggestions carefully.

**Review by sneaxiy — DISMISSED** [review]

**Comment by yuweivvv** [comment]
补充一次更新：除了三处代码，本 PR 现在还把 `third_party/llm-internal-medicine` 指针从 `d9447ac` 前进到 `734593f`（fast-forward，49 个 commit）。

原因是 callback 新增了 `from internal_medicine.core.metric_families import exclusions_for`。这个 import 在现有的 guarded try 里，所以旧版本库不会让训练崩，但会静默落到「library not importable」的 warning 分支、一个 monitor 都不注册——按文档配了 yaml 却毫无输出，排查方向完全不在这里。所以指针跟着一起动，比留个坑好。

**Review by From00 — APPROVED** [review]
LGTM

**Review by zjjlivein — APPROVED** [review]
